### PR TITLE
(Chore) Skip claim queue while running cypress tests

### PR DESCRIPTION
**Pull request by serdiukov-o-nordwhale**
# Description

- add isE2ERunning to skip queue condition
- refactor claimQueueAPI a bit
# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/server/claimQueue/claimQueueAPI.js [hunk 1]
```diff
@@ -46,21 +46,31 @@ const ClaimQueue = {
...
+    const openSpaces = claimQueueAllowed - totalQueued
     let status = openSpaces > 0 ? 'approved' : 'pending'
-    //if user was added to queue tag him in mautic
-    if (['test', 'development'].includes(conf.env) === false && user.mauticId && status === 'pending')
-      Mautic.addContactsToSegment([user.mauticId], conf.mauticClaimQueueSegment).catch(e => {
+
+    // if user was added to queue tag him in mautic
+    if (!['test', 'development'].includes(conf.env) && user.mauticId && status === 'pending') {
+      try {
+        await Mautic.addContactsToSegment([user.mauticId], conf.mauticClaimQueueSegment)
```
Comment (yaroslav-fedyshyn-nordwhale) [inline src/server/claimQueue/claimQueueAPI.js:66]: mauticClaimQueueSegment -> mauticClaimQueueSegmentId

Comment (serdiukov-o-nordwhale) [reply src/server/claimQueue/claimQueueAPI.js:66]: @yaroslav-fedyshyn-nordwhale please use `+/-` (Suggest) for one-line fixes. It's easy to create fix commit then , directly from web UI

Comment (yaroslav-fedyshyn-nordwhale) [inline src/server/claimQueue/claimQueueAPI.js:66]:
```suggestion
        await Mautic.addContactsToSegment([user.mauticId], conf.mauticClaimQueueSegmentId)
```

File: src/server/claimQueue/claimQueueAPI.js [hunk 2]
```diff
@@ -89,11 +99,20 @@ const setup = (app: Router, storage: StorageAPI) => {
...
+      // if queue is enabled and we're not running cypress tests, enqueueing user
+      if (claimQueueAllowed > 0 && !isE2ERunning) {
+        const queueStatus = await ClaimQueue.enqueue(user, storage, log)
+
+        res.json(queueStatus)
+        return
+      }
+
+      log.debug('claimqueue: skip', { claimQueueAllowed, isE2ERunning })
+      res.json({ ok: 0, queue: { status: 'verified' } })
```
Comment (sirpy) [inline src/server/claimQueue/claimQueueAPI.js:115]:
```suggestion
      res.json({ ok: 1, queue: { status: 'verified' } })
```

**Pull request merged**